build/Shared/HashCodeCombiner.cs

```diff
@@ -79,29 +72,80 @@ internal void AddSequence<T>(IEnumerable<T> sequence)
             }
         }
 
+        internal void AddSequence<T>(T[] array)
+        {
+            if (array != null)
+            {
+                foreach (var item in array)
+                {
+                    AddObject(item);
```

Comment:
> Why the approach for sequence is different from the one for dictionary?

Dictionaries are `IReadOnlyDictionary` or `IDictionary` so their enumerators are going to box anyway; also it sorts them by key before creating the hashcode so two dictionaries with the same items will create the same hashcode (as the two dictionarys would be unordered; and otherwise may produce different hashcodes)

So don't think there is much that can be done :-/

> Would it be better with EnsureInitialize() once and AddInt32(item.GetHashCode()) for every item?

Whoops... yes 😄